src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -68,278 +69,462 @@ def __init__(self):
-                now(tz=TIME_ZONE).format(f"{DATE_TIME_FORMAT}:ss")
-        else:
-            self.export_subdir = export_subdir
+@dataclass
+class SimulationStatusManager:
+    """State of the Simulation class."""
+    paused: bool = False
+    pause_after: duration = None
+    timed_out: bool = False
+    stopped: bool = False
+    sim_status = "initializing"
+    incremental: bool = False
 
-        self.setup_module_name = setup_module_name
-        self.is_stopped = False
-        self._is_incremental = incremental
-        self.live_events = LiveEvents(self.simulation_config)
-        self.kafka_connection = kafka_connection_factory()
-        self.redis_connection = RedisSimulationCommunication(self, redis_job_id, self.live_events)
-        self._simulation_id = redis_job_id
-        self._started_from_cli = redis_job_id is None
+    @property
+    def status(self):
+        """Return status of simulation."""
+        if self.timed_out:
+            return "timed-out"
+        if self.stopped:
+            return "stopped"
+        if self.paused:
+            return "paused"
+        return self.sim_status
 
-        self.run_start = None
-        self.paused_time = None
+    def stop(self):
+        """Stop simulation."""
+        self.stopped = True
 
-        self._load_setup_module()
-        self._init(**self.initial_params, redis_job_id=redis_job_id, enable_bc=enable_bc)
+    @property
+    def finished(self):
+        """Return if simulation has finished."""
+        return self.sim_status == "finished"
 
-        deserialize_events_to_areas(simulation_events, self.area)
+    def toggle_pause(self):
+        """Pause or resume simulation."""
+        if self.finished:
+            return False
+        self.paused = not self.paused
+        return True
 
-        validate_const_settings_for_simulation()
+    def handle_pause_after(self, time_since_start):
+        """Deals with pause-after parameter, which pauses the simulation after some time."""
+        if self.pause_after and time_since_start >= self.pause_after:
+            self.paused = True
+            self.pause_after = None
+
+    def handle_pause_timeout(self, tick_time_counter):
+        """
+        Deals with the case that the pause time exceeds the simulation timeout, in which case the
+        simulation should be stopped.
+        """
+        if time() - tick_time_counter > SIMULATION_PAUSE_TIMEOUT:
+            self.timed_out = True
+            self.stopped = True
+            self.paused = False
+        if self.stopped:
+            self.paused = False
+
+    def handle_incremental_mode(self):
+        """
+        Handle incremental paused mode, where simulation is paused after each market slot.
+        """
+        if self.incremental:
+            self.paused = True
+
+
+@dataclass
+class SimulationTimeManager:
+    """Handles simulation time management."""
+    start_time: datetime = now(tz=TIME_ZONE)
+    tick_time_counter: datetime = time()
```

Comment:
```suggestion
    tick_time_counter: DateTime = time()
```
In order to make sure that pendulum datetime is used here and the `.add()` in `handle_slowdown_and_realtime` works. Thanks!

Reply:
Fixing, thanks!